src/consumer/mod.rs

```diff
@@ -42,7 +42,7 @@ pub trait ConsumerContext: ClientContext {
         &self,
         native_client: &NativeClient,
         err: RDKafkaRespErr,
-        tpl: &TopicPartitionList,
+        tpl: &mut TopicPartitionList,
```

Comment:
It looks to me like the current code is correct.

[`rd_kafka_assign`](https://github.com/edenhill/librdkafka/blob/bd74dc31ba6edf431749b4ad5a00aa4fc78882fc/src/rdkafka_subscription.c#L97) is expecting a `const` pointer. The code you pointed out uses code that mutates the value pointed to the pointer, however the pointer itself seems constant.

Is the current code causing issues when using it?

Reply:
It's not possible to mutate &TPL now, but it's not forbidden in librdkafka.
The definition of the callback: https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L1599

Reply:
Fair enough, thank you for the patch 🙂